### ISSUE TITLE
Rule request: overriden declaration in extension

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1884]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Adding `override` for Swift-only declarations is already not possible. It's only possible for `@objc` declarations.

From [Apple's Programming with Objective-C](https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/CustomizingExistingClasses/CustomizingExistingClasses.html#//apple_ref/doc/uid/TP40011210-CH6-SW1)

> Because the methods declared in a category are added to an existing class, you need to be very careful about method names.
>
> If the name of a method declared in a category is the same as a method in the original class, or a method in another category on the same class (or even a superclass), the behavior is undefined as to which method implementation is used at runtime.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// should trigger
extension Foo {
  override var bar: String { return "" }

  override func baz() {}
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just severity.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

I think it should be enabled by default, but I'm afraid that this might cause a lot of violations that are not easily fixed.

---

I have an initial implementation for this already, I just need some time to clean it up.

**Comment by marcelofabri** [comment]
cc @AliSoftware who originally pitched this idea

**Comment by AliSoftware** [comment]
❤️
